io.github.atari800.Atari800

PatrickJMaloney opened the pull request:
New app submission for Atari 800 Emulator.  Home page atari800.github.io.

Please review and advise.

### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
        Not a direct contributor.  Just a user.  Discussion with developers here:  https://github.com/atari800/atari800/issues/71
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
        I do not own the domain...github.io
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
        Patches to SDL 1.2 *not* submitted.  The patch fixes a known issue due to changes in X11, but has not been integrated by the SDL developers for whatever reason.
[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,54 @@
+      - install -Dm 0644 atari800-32x32.png ${FLATPAK_DEST}/share/icons/hicolor/32x32/apps/${FLATPAK_ID}.png
+      - install -Dm 0644 atari800-48x48.png ${FLATPAK_DEST}/share/icons/hicolor/48x48/apps/${FLATPAK_ID}.png
+      - install -Dm 0644 atari800-64x64.png ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png
+      - install -Dm 0644 atari800-128x128.png ${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png
```
only 128x128 is required. so if the icons don't come from upstream limit the number here. (they are missing btw see my other comment)

However, upstream has https://github.com/atari800/atari800/blob/master/data/atari2.svg . and even https://github.com/atari800/atari800/blob/master/data/atari2.png (128x128) This should be enough

PatrickJMaloney replied:
Switched icon to use the SVG at that URL.

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,54 @@
+      - --enable-pngcodec
+      - --with-mp3=lame
+    post-install:
+      - install -Dm 0644 ${FLATPAK_ID}.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
```
this file is missing, but again it is upstream https://github.com/atari800/atari800/blob/master/debian/atari800.desktop

PatrickJMaloney replied:
Instructions said to only commit the manifest.  Will commit the rest.

hfiguiere replied:
I mentionned it's in the upstream repo. you should use that one instead.

PatrickJMaloney replied:
When I try to use that file, I get a warning about the icon name not matching the flatpak app id.  Another comment mentioned using desktop-file-edit to modify the .desktop file entries.  I cannot find desktop-file-edit documented anywhere.

PatrickJMaloney replied:
What you're telling me to do breaks the build.  If I cannot update that .desktop file with the correct icon name, this isn't going to work.

hfiguiere replied:
No. I'm telling use the one that ship with upstream. If it needs fixing because it's broken upstream it can be fixed.

`man desktop-file-edit` for the documentation.

PatrickJMaloney replied:
I used sed to edit the file to put in the correct icon name.

hfiguiere replied:
please don't use sed for that. that's what `destkop-file-edit` is for.

PatrickJMaloney replied:
Updated to use desktop-file-edit.

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,54 @@
+        url: https://libsdl.org/release/SDL-1.2.15.tar.gz
+        sha256: d6d316a793e5e348155f0dd93b979798933fb98aa1edebcc108829d6474aad00
+      - type: patch
+        path: patch/sdlpatch.txt
```
patch is missing.

PatrickJMaloney replied:
Instructions said to only commit the manifest.  Will commit the rest.

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,54 @@
+runtime-version: '6.4'
+sdk: org.kde.Sdk
+command: atari800
+build-options:
+  no-debuginfo: true
+  strip: true
```
```suggestion
```

strip is default. and debuginfo is extracted.

PatrickJMaloney replied:
Removed.

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,54 @@
+app-id: io.github.atari800.Atari800
+runtime: org.kde.Platform
```
I don't see any traces of Qt upstream. Are you sure you need the KDE runtime?

PatrickJMaloney replied:
Switched back to freedesktop.

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,54 @@
+        url: https://github.com/atari800/atari800.git
+        tag: ATARI800_5_0_0
+        commit: c0f94a7bf15543d8feb87ba046c456087c9d8227
+      - type: dir
+        path: misc
+      - type: dir
+        path: icons
```
1. neither nor directories exist
2. please reference

PatrickJMaloney replied:
Instructions said to only commit the manifest.  Will commit the rest.

hfiguiere reviewed (changes requested):
right now this can't even build.

PatrickJMaloney commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,54 @@
+  - --socket=wayland
+  - --device=dri
+  - --socket=pulseaudio
+  - --filesystem=home
```
I need to create and rw ~/.atari800.cfg.  I originally had:
--filesystem=~/.atari800.cfg
--filesystem=xdg-documents

However, this only worked if ~/.atari800.cfg already existed.  If it was not there, it did not get created.  I messed around with --persist which looked like it might be the solution, but could not get that to work.  So, I just used --filesystem=home instead.

How do I allow access to the one file in ~/ and also allow it to be created?

barthalion replied:
This depends on the app. Possibly `--persist=.` would do what you want, but it needs additional tests to see how it interfacts with opening files.

PatrickJMaloney replied:
"--persist=." worked.  Updated yml.  Thanks.

PatrickJMaloney commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,54 @@
+        path: misc
+      - type: dir
+        path: icons
+finish-args:
```
Moved.

PatrickJMaloney commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,54 @@
+      - install -Dm 0644 atari800-64x64.png ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png
+      - install -Dm 0644 atari800-128x128.png ${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png
+      - install -Dm 0644 atari800-256x256.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+      - rm ${FLATPAK_DEST}/bin/sdl-config
+      - rm -rf ${FLATPAK_DEST}/include
+      - rm -rf ${FLATPAK_DEST}/lib/pkgconfig
+      - rm ${FLATPAK_DEST}/lib/*.a
+      - rm ${FLATPAK_DEST}/lib/*.la
```
Moved.

PatrickJMaloney commented:
Updated manifest and added additional files.  Waiting for response on --filesystem=home.

I changed --device to --device=all as that seems to be needed for joysticks to work.

I kept my .desktop file as it has an icon name which is the flatpak id.

barthalion commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,47 @@
+        url: https://github.com/atari800/atari800.git
+        tag: ATARI800_5_0_0
+        commit: c0f94a7bf15543d8feb87ba046c456087c9d8227
+      - type: dir
+        path: misc
```
Ideally icon, appdata and desktop files should come from https://github.com/atari800/atari800.git If e.g. desktop file is already there but needs changes, `desktop-file-edit` can be used to modify specific fields during build.

hfiguiere replied:
there is a svg and png there as I mentionned in the review.

hfiguiere replied:
also it's better to reference files individually than use `dir`.

PatrickJMaloney replied:
Removed 'dir'.

barthalion commented:
bot, build io.github.atari800.Atari800

barthalion commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,47 @@
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=all	
```
There's a trailing tab on this line.

PatrickJMaloney replied:
Fixed.

barthalion commented:
Err, approved it by mistake. Obviously there are still some fixes to be done.

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,47 @@
+        sha256: d6d316a793e5e348155f0dd93b979798933fb98aa1edebcc108829d6474aad00
+      - type: patch
+        path: patch/sdlpatch.txt
+        strip-components: 1
```
```suggestion
```

this is the default.

PatrickJMaloney replied:
Removed.

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,47 @@
+    post-install:
+      - install -Dm 0644 ${FLATPAK_ID}.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm 0644 ${FLATPAK_ID}.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      # Access SVG icon inside local copy of git repo
```
```suggestion
```

no need to state the obvious.

PatrickJMaloney replied:
Removed.

hfiguiere commented on misc/io.github.atari800.Atari800.metainfo.xml:
```diff
@@ -0,0 +1,36 @@
+  <provides>
+    <binary>atari800</binary>
+  </provides>
+  <content_rating type="oars-1.0" />
```
```suggestion
  <content_rating type="oars-1.1" />
```

PatrickJMaloney replied:
Updated.

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,50 @@
+        tag: ATARI800_5_0_0
+        commit: c0f94a7bf15543d8feb87ba046c456087c9d8227
+      - type: file
+        url: https://raw.githubusercontent.com/atari800/atari800/master/debian/atari800.desktop
```
Never get a file from a branch. Reference a specific tag or commit. Always.

PatrickJMaloney replied:
Done.

hfiguiere commented:
bot, build io.github.atari800.Atari800

hfiguiere commented on io.github.atari800.Atari800.yml:
```diff
@@ -0,0 +1,50 @@
+  - name: libsdl
+    sources:
+      - type: archive
+        url: https://libsdl.org/release/SDL-1.2.15.tar.gz
```
I think you should use the shared-modules to build this:

https://github.com/flathub/shared-modules/blob/master/SDL/SDL-1.2.15.json

The top-level README explain how to use it.

PatrickJMaloney replied:
I don't have access to git to commit a change to add the submodule.

PatrickJMaloney replied:
Can I limit this to x86_64 only?

hfiguiere replied:
why?

PatrickJMaloney replied:
It's failing on the arm64 build, right?

hfiguiere replied:
you refused to do what I recommended.

PatrickJMaloney replied:
I can't do that because I don't have access to the git repo.  You are deliberately trying to fail this.  And, you know what?  You win.  I give up.  I have no more time to waste on this.  I got it working on my Steam Deck and that's all I wanted.  F everybody else.
